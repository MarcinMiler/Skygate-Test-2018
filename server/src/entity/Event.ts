import { Entity, PrimaryGeneratedColumn, Column, BaseEntity } from 'typeorm'

@Entity()
export class Event extends BaseEntity {
    @PrimaryGeneratedColumn() id: number

    @Column() title: string

    @Column() description: string

    @Column() organizer: string

    @Column() location: string

    @Column({ type: 'float', nullable: true })
    lat: number

    @Column({ type: 'float', nullable: true })
    lng: number

    @Column() startDate: string

    @Column() endDate: string

    @Column() photo: string

    @Column() category: string
}
