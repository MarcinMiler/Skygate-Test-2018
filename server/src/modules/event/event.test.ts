import { Connection } from 'typeorm'
import { createTestConn } from '../../testUtils/createTestConn'
import { User } from '../../entity/User'

let conn: Connection
beforeAll(async () => {
    conn = await createTestConn()
})

afterAll(async () => {
    conn.close()
})

describe('It should work', () => {
    it('works', async () => {
        await User.create({
            firstName: 'Marcin'
        }).save()

        const u = await User.findOne(1)
        expect(u).toMatchObject({ id: 1, firstName: 'Marcin' })
    })
})
