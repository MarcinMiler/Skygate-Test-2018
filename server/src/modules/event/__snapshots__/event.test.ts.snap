// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return event 1`] = `
Object {
  "event": Object {
    "category": "Music",
    "description": "Some event",
    "endDate": "11.07.2018 20:00",
    "id": 1,
    "lat": 50.5,
    "lng": 35.5,
    "location": "Warsaw",
    "organizer": "Somelab",
    "photo": "photo.jpeg",
    "startDate": "11.07.2018 10:30",
    "title": "Color Festival",
  },
}
`;
